tests/unit-tests/dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -714,6 +958,13 @@ void verify_bool_union(DDS::DynamicType_var dt, const DataView& expected_cdr)
   EXPECT_EQ(ret, DDS::RETCODE_OK);
   ret = data.set_boolean_value(15, CORBA::Boolean(false));
   EXPECT_EQ(ret, DDS::RETCODE_OK);
+
+  CORBA::Long disc_val;
+  EXPECT_EQ(DDS::RETCODE_OK, data.get_int32_value(disc_val, XTypes::DISCRIMINATOR_ID));
+  EXPECT_EQ(E_BOOL, disc_val);
+  CORBA::Boolean bool_val;
+  EXPECT_EQ(DDS::RETCODE_OK, data.get_boolean_value(bool_val, 15));
+  EXPECT_EQ(CORBA::Boolean(false), bool_val);
```

Comment:
I'm not entirely sure what this C++03 error is about, but you could just change this to `EXPECT_FALSE(bool_val)` since `CORBA::Boolean` is always going to be the same thing as `bool`.